### PR TITLE
Added `wasm32` target

### PR DESCRIPTION
**Pull request by catap**
It closes https://github.com/thepowersgang/mrustc/issues/234

### CODE REVIEW

File: src/trans/target.cpp [hunk 1]
```diff
@@ -59,6 +59,13 @@ const TargetArch ARCH_POWERPC64LE = {
     { /*atomic(u8)=*/true, true, true, true,  true },
     TargetArch::Alignments(2, 4, 8, 16, 4, 8, 8)
 };
+// see https://webassembly.org/docs/portability/
+const TargetArch ARCH_WASM32 = {
+    "wasm32",
+    32, false,
+    { /*atomic(u8)=*/true, true, true, false,  true },
```
Comment (bjorn3) [inline src/trans/target.cpp:66]: Wasm32 supports 64bit atomics with the atomics extension. Without it doesn't support any atomics at all.

File: src/trans/target.cpp [hunk 2]
```diff
@@ -59,6 +59,13 @@ const TargetArch ARCH_POWERPC64LE = {
...
+    TargetArch::Alignments(2, 4, 4, 4, 4, 4, 4)
```
Comment (bjorn3) [inline src/trans/target.cpp:67]: This should be 2, 4, 8, 8, 4, 8, 4.

File: src/trans/target.cpp (hunk 1 above)
Comment (catap) [reply src/trans/target.cpp:66]: @bjorn3 thanks

File: src/trans/target.cpp (hunk 2 above)
Comment (catap) [reply src/trans/target.cpp:67]: @bjorn3 thanks

**Comment by thepowersgang** [comment]
Unless you've tested that this 1. fixes the stated error, and 2. works (i.e. can compile working code), I'm not willing to merge it.

**Comment by catap** [comment]
@thepowersgang after applying this patch I've overstep https://github.com/thepowersgang/mrustc/issues/234.

**Comment by catap** [comment]
Regarding working code: I haven't tested it.

**Comment by catap** [comment]
This PR is required and used by https://github.com/macports/macports-ports/pull/13817

**Comment by thepowersgang** [comment]
> This PR is required and used by [macports/macports-ports#13817](https://github.com/macports/macports-ports/pull/13817)

Citation needed. It's very unlikely that wasm is needed to compile for macos.

**Comment by catap** [comment]
I can build without this PR. Seems that an issue was introduced by another issue.

Shall I close this PR for now or would you like to keep it?

**Pull request closed without merging**